client/verta/verta/tracking/entities/_experimentrun.py

```diff
@@ -1305,17 +1305,15 @@ def log_model(self, model, custom_modules=None, model_api=None, artifacts=None,
                     sorted(unlogged_artifact_keys)))
 
         # serialize model
-        try:
-            extension = _artifact_utils.get_file_ext(model)
-        except (TypeError, ValueError):
-            extension = None
         _utils.THREAD_LOCALS.active_experiment_run = self
         try:
             serialized_model, method, model_type = _artifact_utils.serialize_model(
                 model)
         finally:
             _utils.THREAD_LOCALS.active_experiment_run = None
-        if extension is None:
+        try:
+            extension = _artifact_utils.get_file_ext(serialized_model)
+        except (TypeError, ValueError):
             extension = _artifact_utils.ext_from_method(method)
```

Comment:
I noticed that `ModelVersion` doesn't experience this bug so to mirror that logic ([_modelversion.py#L306-L311](https://github.com/VertaAI/modeldb/blob/b8c9add/client/verta/verta/registry/entities/_modelversion.py#L306-L311)), I'm getting rid of the pre-serialization `get_file_ext()` step and moving it to after.